osh/builtin.py

```diff
@@ -731,6 +732,27 @@ def __call__(self, cmd_val):
     else:
       raise args.UsageError('Too many arguments')
 
+    # Clear all history
+    if arg.c:
+      readline_mod.clear_history()
+      return 0
+
+
+    # Delete history entry by id number
+    if arg.d:
+      try:
+        cmd_index = int(rest[0]) - 1  # readline is indexed from 1
```

Comment:
This has the issue I mentioned on Zulip -- `int()` can raise ValueError.

The idiomatic way to do it is:

HISTORY_SPEC.ShortFlag('-d', args.Int)

And then arg.d is the integer itself.

This also makes `-d4` (no space) and `-d 4` (space) work, since the arg parser understands that.

Reply:
`-d4` and `-d 4` working is worth losing the more-specific error message.